diff -Nuar a/kernel/nv-drm.c b/kernel/nv-drm.c
--- a/kernel/nv-drm.c
+++ b/kernel/nv-drm.c 
@@ -128,6 +128,10 @@
     .gem_prime_vmap = nv_gem_prime_vmap,
     .gem_prime_vunmap = nv_gem_prime_vunmap,
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 18, 0)
+    .set_busid = drm_pci_set_busid,
+#endif
+
     .name = "nvidia-drm",
     .desc = "NVIDIA DRM driver",
     .date = "20130102",
diff -Nuar a/kernel/nv-linux.h b/kernel/nv-linux.h
--- a/kernel/nv-linux.h
+++ b/kernel/nv-linux.h
@@ -2000,7 +2000,7 @@
 #if defined(NV_FILE_HAS_INODE)
 #define NV_FILE_INODE(file) (file)->f_inode
 #else
-#define NV_FILE_INODE(file) (file)->f_dentry->d_inode
+#define NV_FILE_INODE(file) (file)->f_path.f_dentry->d_inode
 #endif
 
 /* Stub out UVM in multi-RM builds */
