diff --git a/volti b/volti
index d8f06bd..25d7ca9 100755
--- a/volti
+++ b/volti
@@ -19,15 +19,8 @@
 
 import os
 import sys
-
-try:
-    import dbus
-    import dbus._version
-    from dbus.mainloop.glib import DBusGMainLoop
-    assert dbus.version >= (0, 80, 0)
-except ImportError, AssertionError:
-    sys.stderr.write("This program needs dbus-python 0.80.0 or higher\nExiting\n")
-    sys.exit(1)
+import dbus
+from dbus.mainloop.glib import DBusGMainLoop
 
 if os.path.isdir(os.path.join(".","src")) and os.path.isfile(
         os.path.join(".","setup.py")):

